### PR TITLE
Refactoring: moving uuf.internal.core.*' packages to 'uuf.internal', renaming Logger objects to 'LOGGER'

### PR DESCRIPTION
**Pull request by this**
- Moving `org.wso2.carbon.uuf.internal.core.auth` package to `org.wso2.carbon.uuf.internal.auth`
- Moving `org.wso2.carbon.uuf.internal.core.deployment` package to `org.wso2.carbon.uuf.internal.deployment`
- Renamed all the Logger objects from `log` to `LOGGER`
- Instead of concatenation, used SLF4J [parameterized logging messages](http://slf4j.org/faq.html#logging_performance).

### CODE REVIEW

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/deployment/ClassLoaderProvider.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.uuf.internal.deployment;
+
+import org.wso2.carbon.uuf.api.reference.ComponentReference;
+
+public interface ClassLoaderProvider {
```
Comment (Kishanthan) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/deployment/ClassLoaderProvider.java]: Class level comment is missing.

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/deployment/AppDeployer.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * under the License.
+ */
+
+package org.wso2.carbon.uuf.internal.deployment;
+
+import org.wso2.carbon.uuf.core.App;
+
+import java.util.Set;
+
+public interface AppDeployer {
```
Comment (Kishanthan) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/deployment/AppDeployer.java]: Class level and interface method level java docs are missing.

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/deployment/DeploymentNotifier.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * under the License.
+ */
+
+package org.wso2.carbon.uuf.internal.deployment;
+
+import org.wso2.carbon.uuf.spi.HttpConnector;
+
+import java.util.Set;
+
+public abstract class DeploymentNotifier {
```
Comment (Kishanthan) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/deployment/DeploymentNotifier.java]: Class level java doc comment is missing.

File: samples/osgi-services/org.wso2.carbon.uuf.sample.pets-store.bundle/src/main/java/org/wso2/carbon/uuf/sample/petsstore/bundle/internal/Activator.java [hunk 4]
```diff
@@ -25,20 +25,16 @@
...
-    private static final Logger log = LoggerFactory.getLogger(Activator.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(Activator.class);
 
     @Override
     public void start(BundleContext bundleContext) throws Exception {
         bundleContext.registerService(PetsStoreService.class.getName(), new PetsManagerImpl(), null);
-        if (log.isDebugEnabled()) {
-            log.debug("Pet-Store BE Bundle Started.");
-        }
+        LOGGER.debug("Pet-Store BE Bundle started.");
```
Comment (Kishanthan) [inline samples/osgi-services/org.wso2.carbon.uuf.sample.pets-store.bundle/src/main/java/org/wso2/carbon/uuf/sample/petsstore/bundle/internal/Activator.java]: We still need to check isDebug enabled as this does not use marker

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/deployment/ClassLoaderProvider.java (hunk 1 above)
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/deployment/ClassLoaderProvider.java]: fixed in commit fd2881435e3ca34794f055620721b62d79cbb327

File: samples/osgi-services/org.wso2.carbon.uuf.sample.pets-store.bundle/src/main/java/org/wso2/carbon/uuf/sample/petsstore/bundle/internal/Activator.java (hunk 4 above)
Comment (this) [inline samples/osgi-services/org.wso2.carbon.uuf.sample.pets-store.bundle/src/main/java/org/wso2/carbon/uuf/sample/petsstore/bundle/internal/Activator.java]: fixed in commit 047ae8c67f8fb5aa507473d8637fe3073267491e

**Pull request merged**